web/src/transmission.js

```diff
@@ -961,22 +999,26 @@ TODO: fix this when notifications get fixed
     const trackers = this._getTrackerCounts();
     const sitenames = Object.keys(trackers).sort();
 
-    // build the new html
-    let string = '';
-    string += this.filterTracker
-      ? '<option value="all">All</option>'
-      : '<option value="all" selected="selected">All</option>';
-    for (const sitename of sitenames) {
-      string += `<option value="${sitename}"`;
-      if (sitename === this.filterTracker) {
-        string += ' selected="selected"';
+    // Update select box only when list of trackers has changed
+    if (
+      sitenames.length !== this.oldTrackers.length ||
+      sitenames.some((x) => !this.oldTrackers.includes(x))
```

Comment:
Since `sitenames` and `this.oldTrackers` are sorted, this line can be changed like this for a bit more performance.

```suggestion
      sitenames.some((ele, idx) => ele !== this.oldTrackers[idx])
```

Reply:
Surprisingly no error, I know Python would otherwise, thank you for the performance advice! I can see how it can perform faster.